src/pages/[marketplace]/deals/new.tsx

```diff
@@ -0,0 +1,97 @@
+import DealForm, { DealFormInput } from "@components/DealForm";
+import Layout from "@components/Layout";
+import { Link } from "@components/Link";
+import { useCredixClient } from "@credix/credix-client";
+import { PublicKey } from "@solana/web3.js";
+import { numberFormatter, toProgramAmount } from "@utils/format.utils";
+import Big from "big.js";
+import message from "message";
+import { useRouter } from "next/router";
+import { NextPageWithLayout } from "pages/_app";
+import React, { ReactElement, useEffect } from "react";
+import { useStore } from "state/useStore";
+
+const New: NextPageWithLayout = () => {
+	const router = useRouter();
+	const { marketplace } = router.query;
+	const client = useCredixClient();
+	const fetchMarket = useStore((state) => state.fetchMarket);
+	const market = useStore((state) => state.market);
+
+	useEffect(() => {
+		fetchMarket(client, marketplace as string);
+	}, [client, fetchMarket, marketplace]);
+
+	const onSubmit = async ({
+		principal,
+		financingFee,
+		timeToMaturity,
+		borrower,
+		dealName,
+	}: DealFormInput) => {
+		const formattedPrincipal = numberFormatter.format(principal);
+		const hide = message.loading({ content: `Creating deal for ${formattedPrincipal} USDC` });
+		const borrowerPK = new PublicKey(borrower);
+
+		try {
+			const credixPass = await market.fetchCredixPass(borrowerPK);
+
+			if (!credixPass) {
+				hide();
+				message.error({ content: "No Credix Pass found for given public key" });
+				return;
+			}
+		} catch {
+			hide();
+			message.error({ content: `Failed to get Credix pass for given public key` });
+			return;
+		}
+
+		try {
+			const principalProgramAmount = toProgramAmount(new Big(principal)).toNumber();
+			await market.createDeal(
+				principalProgramAmount,
+				financingFee,
+				timeToMaturity,
+				borrowerPK,
+				dealName
+			);
+			hide();
```

Comment:
if this is the message that says we are creating the deal it should be before the createDeal call?

Reply:
Yes, it is displayed before the `createDeal` call. 

```typescript
const hide = message.loading({ content: `Creating deal for ${formattedPrincipal} USDC` });
```
Above code will instantly display the message. `hide()` can then be called to manually hide the message.

Reply:
alright, I misread the code!